include/smack/Prelude.h

```diff
@@ -0,0 +1,61 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+
+#ifndef PRELUDE_H
+#define PRELUDE_H
+
+#include "smack/BoogieAst.h"
+#include "smack/SmackRep.h"
+
+#include <string>
+
+namespace smack {
+
+class Prelude {
+  SmackRep &rep;
+
+  FuncDecl* uninterpretedUnaryOp(int width, std::string name);
+  FuncDecl* uninterpretedBinaryOp(int width, std::string name);
+  FuncDecl* inlineBinaryOp(int width, std::string name, const Expr* body);
+  FuncDecl* builtinBinaryOp(int width, std::string builtinName, std::string name);
+  FuncDecl* bvbuiltinUnaryOp(int width, std::string builtinName, std::string name);
+  FuncDecl* bvbuiltinBinaryOp(int width, std::string builtinName, std::string name);
+  FuncDecl* inlineBinaryComp(int width, std::string name, const Expr* body);
+  FuncDecl* inlineBinaryBvComp(int width, std::string name, const Expr* body);
+  FuncDecl* bvbuiltinBinaryComp(int width, std::string builtinName, std::string name);
+  FuncDecl* inlineBinaryPred(int width, std::string name);
+  FuncDecl* inlineBinaryBvPred(int width, std::string name);
+  FuncDecl* safeLoadOp(int width, const Expr* body);
+  FuncDecl* safeStoreOp(int width, const Expr* body);
+  FuncDecl* safeLoadBvOp(int width, const Expr* body);
+  FuncDecl* safeStoreBvOp(int width, const Expr* body);
+  FuncDecl* inlineBvbuiltinBinarySelect(int width, std::string name, std::string name2);
+  FuncDecl* inlineBvbuiltinBinaryPred(int width, std::string name);
+  FuncDecl* inlineConversion(int width, int width2, std::string name, const Expr* body);
+  FuncDecl* truncBvOp(int width, int width2);
+  FuncDecl* truncOp(int width, int width2);
+
+  std::string getIntegerTypeName(int width);
+  std::string getBvTypeName(int width);
+
+  std::string makeVarName(int i);
```

Comment:
Do these functions need to be members of `Prelude` class? Are they used by other classes?

Reply:
@shaobo-he is I think right here. Since these are "free" helper functions (they don't access members of `Prelude`), it would be better to just put their implementations in the cpp file, without having them be private members of the `Prelude` class. See: https://stackoverflow.com/questions/1145482/should-c-class-helper-functions-be-members-free-or-anon-namespace-free